### PR TITLE
refactor: net: subnet lookup: use single-result LookupHost()

### PR DESCRIPTION
**Pull request by theStack**
plus describe single IP subnet case for more clarity

### CODE REVIEW

**Comment by MarcoFalke** [comment]
This is refactor, right?

**Comment by theStack** [comment]
> 
> 
> This is refactor, right?

True, I added the `refactor` prefix to the pull request title and commit message.

**Comment by practicalswift** [comment]
FWIW - `LookupHost(…)` usages:

```
CConnman::ThreadDNSAddressSeed() → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
main → AppInit(int, char**) → AppInitMain(InitInterfaces&) → InitHTTPServer() → LookupHost(char const*, CNetAddr&, bool)
main → AppInit(int, char**) → AppInitMain(InitInterfaces&) → InitHTTPServer() → LookupHost(char const*, CNetAddr&, bool) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
main → AppInit(int, char**) → AppInitMain(InitInterfaces&) → InitHTTPServer() → LookupSubNet(char const*, CSubNet&) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
main → AppInit(int, char**) → AppInitMain(InitInterfaces&) → LookupSubNet(char const*, CSubNet&) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
setban(JSONRPCRequest const&) → LookupHost(char const*, CNetAddr&, bool)
setban(JSONRPCRequest const&) → LookupHost(char const*, CNetAddr&, bool) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
setban(JSONRPCRequest const&) → LookupSubNet(char const*, CSubNet&) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
ThreadMapPort() → LookupHost(char const*, CNetAddr&, bool)
ThreadMapPort() → LookupHost(char const*, CNetAddr&, bool) → LookupHost(char const*, std::vector<CNetAddr>&, unsigned int, bool)
```

**Comment by theStack** [comment]
> 
> 
> as this concerns DNS: @TheBlueMatt

Note that no real lookup (resolving) is performed in `LookupSubNet()`, as the calls to `LookupHost()` pass the flag `bool fAllowLookup` = `false`. To quote the further called function `LookupIntern()`:

https://github.com/bitcoin/bitcoin/blob/c34b88620dc8435b83e6744895f2ecd3c9ec8de7/src/netbase.cpp#L91-L93

Hence the calls to `LookupHost()` in `LookupSubNet()` are only used to parse an IP-Address string and convert it to a `CNetAddr` structure.

**Comment by theStack** [comment]
Rebased.

File: src/netbase.cpp [hunk 1]
```diff
@@ -826,14 +826,11 @@ bool LookupSubNet(const std::string& strSubnet, CSubNet& ret)
         return false;
     }
     size_t slash = strSubnet.find_last_of('/');
-    std::vector<CNetAddr> vIP;
+    CNetAddr network, netmask;
```
Comment (practicalswift) [inline src/netbase.cpp:829]: The scope of `netmask` could be reduce to make review somewhat easier?

**Comment by theStack** [comment]
Rebased on master again and limited the scope of `netmask`, [as suggested by practicalswift](https://github.com/bitcoin/bitcoin/pull/17160#discussion_r594608228).

File: src/netbase.cpp (hunk 1 above)
Comment (theStack) [reply src/netbase.cpp]: Good idea! Done.

**Review by jonatack — APPROVED** [review]
This looks like it is one of your first pulls!

utACK a989f98d240a84b5c798252acaa4a316ac711189 the patch rebases cleanly to master, the debug build is green, and it is essentially the same patch as c8991f0251dd2a modulo local variable naming, braced initialization, and a comment

**Review by vasild — APPROVED** [review]
ACK a989f98d240a84b5c798252acaa4a316ac711189

**Pull request merged**